libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -31,9 +31,75 @@
 #include "libavutil/avassert.h"
 #include "dnn_backend_native_layer_pad.h"
 #include "dnn_backend_native_layer_maximum.h"
+#include "libtf_wrapper.h"
+
 
 #include <tensorflow/c/c_api.h>
 
+#ifdef _WIN32
+#include <windows.h>
+#define TF_LOAD_FUNC(path) LoadLibrary(path)
+#define TF_SYM_FUNC(lib, sym) GetProcAddress(lib, sym)
+#define TF_ERROR_FUNC() GetLastError()
+#define TF_FREE_FUNC(lib) FreeLibrary(lib)
+#define TF_LIBNAME "tensorflow.dll"
+#else
+#include <dlfcn.h>
+#define TF_LOAD_FUNC(path) dlopen(path, RTLD_LAZY | RTLD_GLOBAL)
+#define TF_SYM_FUNC(lib, sym) dlsym(lib, sym)
+#define TF_ERROR_FUNC() dlerror()
+#define TF_FREE_FUNC(lib) dlclose(lib)
+#define TF_LIBNAME "libtensorflow.so"
+#endif
+
+#define FN_LIST(MACRO)					    \
+    MACRO(TF_SessionRun)				    \
+    MACRO(TF_GetCode)					    \
+    MACRO(TF_Dim)					    \
+    MACRO(TF_TensorData)				    \
+    MACRO(TF_TensorType)				    \
+    MACRO(TF_CloseSession)				    \
+    MACRO(TF_NewSession)				    \
+    MACRO(TF_DeleteSession)				    \
+    MACRO(TF_DeleteStatus)				    \
+    MACRO(TF_DeleteGraph)				    \
+    MACRO(TF_NewGraph)					    \
+    MACRO(TF_DeleteTensor)				    \
+    MACRO(TF_GraphOperationByName)			    \
+    MACRO(TF_OperationOutputType)			    \
+    MACRO(TF_NewStatus)					    \
+    MACRO(TF_GraphGetTensorShape)			    \
+    MACRO(TF_NewOperation)				    \
+    MACRO(TF_SetAttrType)				    \
+    MACRO(TF_SetAttrShape)				    \
+    MACRO(TF_FinishOperation)				    \
+    MACRO(TF_AllocateTensor)				    \
+    MACRO(TF_SetAttrTensor)				    \
+    MACRO(TF_AddInput)					    \
+    MACRO(TF_SetAttrInt)				    \
+    MACRO(TF_DataTypeSize)				    \
+    MACRO(TF_SetAttrString)				    \
+    MACRO(TF_SetAttrIntList)				    \
+    MACRO(TF_NewSessionOptions)				    \
+    MACRO(TF_SetConfig)					    \
+    MACRO(TF_DeleteSessionOptions)			    \
+    MACRO(TF_NewImportGraphDefOptions)			    \
+    MACRO(TF_ImportGraphDefOptionsSetDefaultDevice)	    \
+    MACRO(TF_GraphImportGraphDef)			    \
+    MACRO(TF_DeleteImportGraphDefOptions)		    \
+    MACRO(TF_DeleteBuffer)				    \
+    MACRO(TF_NewBuffer)					    \
+    MACRO(TF_Version)
+
+#define TF_LOAD_SYMBOL(sym) tf_model->tffns->sym = (PFN_##sym)TF_SYM_FUNC(tf_model->libtensorflow, #sym);
+
+#define PFN_DEF(name)  \
+    PFN_##name name;
+
+typedef struct TFFunctions {
+    FN_LIST(PFN_DEF)
+} TFFunctions;
+
```

Comment:
Better to move this whole section to the wrapper header I think.

Reply:
Moved to wrapper header